core/src/main/java/io/confluent/rest/jetty/DoSFilter.java

```diff
@@ -1238,7 +1233,7 @@ public OverLimit isRateExceeded(long now) {
             // rate limit is never exceeded, but we keep track of the request timestamps
             // so that we know whether there was recent activity on this tracker
             // and whether it should be expired
-            synchronized (this) {
+            try (AutoLock l = _lock.lock()) {
```

Comment:
Why not stick with the synchronized block?

Reply:
we were trying to find some overlap w/ the Jetty 12 utils. It seems Jetty 12 decided to use this AutoLock class throughout. This DosFilter class takes the base logic of the Jetty 9.4.53 version, but makes changes such that it is compatible w/ Jetty 12